textgrad/engine/openai.py

```diff
@@ -96,3 +96,30 @@ def generate_with_messages(self, messages, temperature=0, max_tokens=2000, top_p
     def __call__(self, prompt, **kwargs):
         return self.generate(prompt, **kwargs)
 
+class AzureChatOpenAI(ChatOpenAI):
+    def __init__(
+        self,
+        model_string="gpt-35-turbo",
+        system_prompt=ChatOpenAI.DEFAULT_SYSTEM_PROMPT,
+        **kwargs):
+        """
+        :param model_string:
+        :param system_prompt:
+        """
+        root = platformdirs.user_cache_dir("textgrad")
+        cache_path = os.path.join(root, f"cache_azure_{model_string}.db")  # Changed cache path to differentiate from OpenAI cache
+
+        super().__init__(cache_path=cache_path, system_prompt=system_prompt, **kwargs)
```

Comment:
Won't this call `ChatOpenAI.__init__()` and so redefine `cache_path` as         `cache_path = os.path.join(root, f"cache_openai_{model_string}.db")`?

Reply:
thanks! may need to change ChatOpenAI too. But I don't want to introduce too many changes. Please advice how to fix it.

Reply:
Yep; needs to be fixed separately. 

https://github.com/zou-group/textgrad/pull/20#issuecomment-2181754251

Reply:
FWIW I use `super(ChatOpenAI, self).__init__(cache_path=cache_path)` locally.